boot/seal_test.go

```diff
@@ -148,6 +153,153 @@ func (s *sealSuite) TestSealKeyToModeenv(c *C) {
 	}
 }
 
+func (s *sealSuite) TestResealKeyToModeenv(c *C) {
+	for _, tc := range []struct {
+		resealErr error
+		err       string
+	}{
+		{resealErr: nil, err: ""},
+		{resealErr: errors.New("reseal error"), err: "cannot reseal the encryption key: reseal error"},
+	} {
+		tmpDir := c.MkDir()
+		dirs.SetRootDir(tmpDir)
+		defer dirs.SetRootDir("")
+
+		err := createMockGrubCfg(filepath.Join(tmpDir, "run/mnt/ubuntu-seed"))
+		c.Assert(err, IsNil)
+
+		err = createMockGrubCfg(filepath.Join(tmpDir, "run/mnt/ubuntu-boot"))
+		c.Assert(err, IsNil)
+
+		modeenv := &boot.Modeenv{
+			CurrentRecoverySystems: []string{"20200825"},
+			CurrentTrustedRecoveryBootAssets: boot.BootAssetsMap{
+				"grubx64.efi": []string{"grub-hash-1"},
+				"bootx64.efi": []string{"shim-hash-1", "shim-hash-2"},
+			},
+
+			CurrentTrustedBootAssets: boot.BootAssetsMap{
+				"grubx64.efi": []string{"run-grub-hash-1", "run-grub-hash-2"},
+			},
+
+			CurrentKernels: []string{"pc-kernel_500.snap", "pc-kernel_600.snap"},
+		}
+
+		// mock asset cache
+		p := filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/bootx64.efi-shim-hash-1")
+		err = os.MkdirAll(filepath.Dir(p), 0755)
+		c.Assert(err, IsNil)
+		err = ioutil.WriteFile(p, nil, 0644)
+		c.Assert(err, IsNil)
+		err = ioutil.WriteFile(filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/bootx64.efi-shim-hash-2"), nil, 0644)
+		c.Assert(err, IsNil)
+		err = ioutil.WriteFile(filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/grubx64.efi-grub-hash-1"), nil, 0644)
+		c.Assert(err, IsNil)
+		err = ioutil.WriteFile(filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/grubx64.efi-run-grub-hash-1"), nil, 0644)
+		c.Assert(err, IsNil)
+		err = ioutil.WriteFile(filepath.Join(tmpDir, "var/lib/snapd/boot-assets/grub/grubx64.efi-run-grub-hash-2"), nil, 0644)
+		c.Assert(err, IsNil)
+
+		model := makeMockUC20Model()
+
+		// set a mock recovery kernel
+		readSystemEssentialCalls := 0
+		restore := boot.MockSeedReadSystemEssential(func(seedDir, label string, essentialTypes []snap.Type, tm timings.Measurer) (*asserts.Model, []*seed.Snap, error) {
+			readSystemEssentialCalls++
+			kernelSnap := &seed.Snap{
+				Path: "/var/lib/snapd/seed/snaps/pc-kernel_1.snap",
+				SideInfo: &snap.SideInfo{
+					Revision: snap.Revision{N: 0},
```

Comment:
this has the same wrong Revision set for it, it should be
```suggestion
					Revision: snap.Revision{N: 1},
```